contracts/strategies/morpho/MetaMorphoStrategy.sol

```diff
@@ -0,0 +1,148 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.10;
+
+import "@openzeppelin/contracts-upgradeable/token/ERC20/utils/SafeERC20Upgradeable.sol";
+import "../../interfaces/IERC20Detailed.sol";
+import "../../interfaces/morpho/IMorphoSupplyVault.sol";
+import "../../interfaces/morpho/IMorphoAaveV2Lens.sol";
+import "../../interfaces/morpho/IMorphoCompoundLens.sol";
+import "../../interfaces/morpho/IRewardsDistributor.sol";
+import "../../interfaces/morpho/IURD.sol";
+import "../../interfaces/morpho/IMetamorphoSnippets.sol";
+import "../ERC4626Strategy.sol";
+
+contract MetaMorphoStrategy is ERC4626Strategy {
+  using SafeERC20Upgradeable for IERC20Detailed;
+
+  //// @notice MORPHO governance token
+  IERC20Detailed internal constant MORPHO = IERC20Detailed(0x9994E35Db50125E0DF82e4c2dde62496CE330999);
+
+  /// @notice reward token address (e.g. MORPHO, COMP, ...)
+  address[] public rewardTokens;
+
+  /// @notice MORPHO transferability
+  bool public morphoTransferable;
+
+  /// @notice Metamorpho snippets contract, used for apr
+  address public mmSnippets;
+
+  /// Initialization
+
+  /// @notice initialize the strategy
+  /// @param _strategyToken strategy token address
+  /// @param _token underlying token address
+  /// @param _owner owner address
+  /// @param _mmSnippets snippets contract address
+  /// @param _rewardTokens array of reward tokens
+  function initialize(
+    address _strategyToken,
+    address _token,
+    address _owner,
+    address _mmSnippets,
+    address[] memory _rewardTokens
+  ) public {
+    _initialize(_strategyToken, _token, _owner);
+
+    // used to fetch apr
+    mmSnippets = _mmSnippets;
+
+    for (uint256 i = 0; i < _rewardTokens.length; i++) {
+      rewardTokens.push(_rewardTokens[i]);
+    }
+  }
+
+  /// External and public methods
+
+  /// @notice redeem the rewards
+  /// @return rewards amount of reward that is deposited to the ` strategy`
+  function redeemRewards(bytes calldata data)
+    public
+    virtual
+    override
+    onlyIdleCDO
+    nonReentrant
+    returns (uint256[] memory rewards)
+  {
+    address[] memory _rewardTokens = rewardTokens;
+    uint256 _rewardsLen = _rewardTokens.length;
+    if (_rewardsLen == 0) {
+      return rewards;
+    }
+
+    rewards = new uint256[](_rewardsLen);
+    bytes[] memory claimDatas = abi.decode(data, (bytes[]));
+    address reward;
+    address rewardDistributor;
+    uint256 claimable; 
+    bytes32[] memory proof;
+
+    for (uint256 i = 0; i < _rewardsLen; i++) {
+      if (claimDatas[i].length == 0) {
+        continue;
+      }
+      (reward, rewardDistributor, claimable, proof) = abi.decode(claimDatas[i], (address, address, uint256, bytes32[]));
+      rewards[i] = _claimReward(IURD(rewardDistributor), reward, claimable, proof);
```

Comment:
note that anyone can claim rewards on behalf of this contract

Reply:
Noted thanks, we will add a way to manage this case, probably by allowing the transfer of rewards in the contract directly in IdleCDO

Reply:
Fixed here https://github.com/Idle-Labs/idle-tranches/pull/83/commits/90dd914919cfec79101833bdbebc0d0132b5c36b